### PR TITLE
adding Async Activity Completion [1454]

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?

https://github.com/temporalio/samples-python/blob/main/hello/hello_async_activity_completion.py is the full sample, but basically inside an activity to mark it as completing async, you would:

```python
# Capture token for later completion
captured_token = activity.info().task_token
activity.raise_complete_async()
```

(this could also work with user-defined activity IDs)

To get a handle to update the activity outside the activity, given a client you would:

```python
handle = my_client.get_async_activity_handle(task_token=captured_token)
```

Then, on that handle, you can call `heartbeat`, `complete`, `fail`, and `report_cancellation` to update the activity.

### CODE REVIEW

File: docs-src/python/how-to-set-asynchronous-activity-completion-in-python.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+id: how-to-set-asynchronous-activity-completion-in-python
+title: How to set an Asynchronous Activity Completion in Python
+sidebar_label: Set an Asynchronous Activity Completion
+description: Set an Asynchronous Activity Completion
```
Comment (djmagee) [inline docs-src/python/how-to-set-asynchronous-activity-completion-in-python.md:5]:
```suggestion
description: To set the completion status of an asynchronous Activity, get the handle of the Activity and call the appropriate method of `AsyncActivityHandle`.
```

**Review by djmagee — COMMENTED** [review]
Please take a close look at my suggestions to ensure that I don't munge your intent.

File: docs-src/python/how-to-set-asynchronous-activity-completion-in-python.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+id: how-to-set-asynchronous-activity-completion-in-python
+title: How to set an Asynchronous Activity Completion in Python
+sidebar_label: Set an Asynchronous Activity Completion
+description: To set the completion status of an asynchronous Activity, get the handle of the Activity and call the appropriate method of `AsyncActivityHandle`.
+tags:
+  - python
+  - how-to
+---
+
+To mark an Activity as completing asynchoronus, do the following.
```
Comment (cretz) [inline docs-src/python/how-to-set-asynchronous-activity-completion-in-python.md:11]:
```suggestion
To mark an Activity as completing asynchoronus, do the following inside the activity:
```

File: docs-src/python/how-to-set-asynchronous-activity-completion-in-python.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+```python
+# Capture token for later completion
+captured_token = activity.info().task_token
+activity.raise_complete_async()
+```
+
+To update an Activity outside the Activity, use [AsyncActivityHandle](https://python.temporal.io/temporalio.client.AsyncActivityHandle.html) to get the handle of the Activity.
```
Comment (cretz) [inline docs-src/python/how-to-set-asynchronous-activity-completion-in-python.md:19]: `AsyncActivityHandle` _is_ the handle, you don't use it to get the handle. You use `get_async_activity_handle` to do that.

File: docs-src/python/how-to-set-asynchronous-activity-completion-in-python.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+```python
+handle = my_client.get_async_activity_handle(task_token=captured_token)
+```
+
+Then, on that handle, you can call the `heartbeat`, `complete`, `fail`, or `report_cancellation` method to update the Activity.
+
+```python
+await handle.complete(f"{input.getting}")
```
Comment (cretz) [inline docs-src/python/how-to-set-asynchronous-activity-completion-in-python.md:28]: `(f"{input.getting}")` isn't very clear here. Can just put `"completion value"` or something.

**Pull request merged**